montblanc/impl/rime/tensorflow/rime_ops/tests/test_zernike.py

```diff
@@ -0,0 +1,773 @@
+import pytest
+import time
+
+import numpy as np
+import tensorflow as tf
+from tensorflow.python.client import device_lib
+import matplotlib.pyplot as plt
+
+from montblanc.impl.rime.tensorflow.tensorflow_ops import zernike
+
+
+@pytest.mark.parametrize("FT, CT", [(np.float32, np.complex64), (np.float64, np.complex128)])
+def test_zernike_xx(gpu_devs, coeff_xx, noll_index_xx, eidos_data_xx, FT, CT):
+    """ Test the Zernike operator """
+    _impl_test_zernike(FT, CT, gpu_devs, coeff_xx, noll_index_xx, 15, eidos_data_xx, 0)
+
+@pytest.mark.parametrize("FT, CT", [(np.float32, np.complex64), (np.float64, np.complex128)])
+def _test_zernike_xy(gpu_devs, coeff_xy, noll_index_xy, eidos_data_xy, FT, CT):
+    """ Test the Zernike operator """
+    _impl_test_zernike(FT, CT, gpu_devs, coeff_xy, noll_index_xy, 8, eidos_data_xy, 1)
+
+@pytest.mark.parametrize("FT, CT", [(np.float32, np.complex64), (np.float64, np.complex128)])
+def _test_zernike_yx(gpu_devs, coeff_yx, noll_index_yx, eidos_data_yx, FT, CT):
+    """ Test the Zernike operator """
+    _impl_test_zernike(FT, CT, gpu_devs, coeff_yx, noll_index_yx, 8, eidos_data_yx, 2)
+
+@pytest.mark.parametrize("FT, CT", [(np.float32, np.complex64), (np.float64, np.complex128)])
+def _test_zernike_yy(gpu_devs, coeff_yy, noll_index_yy, eidos_data_yy, FT, CT):
+    """ Test the Zernike operator """
+    _impl_test_zernike(FT, CT, gpu_devs, coeff_yy, noll_index_yy, 15, eidos_data_yy, 3)
+
+
+def _impl_test_zernike(FT, CT, gpu_devs, coeff_nn, noll_index_nn, thresh, eidos_data_nn, corr_num):
+    """ Implementation of the Zernike operator test """
+    npix = 17
+    nsrc = npix ** 2
+    ntime = 20
+    na = 20
+    nchan = 20
+
+    nx, ny = npix, npix
+    grid = (np.indices((nx, ny), dtype=np.float) - nx//2) * 2 / nx
+    ll, mm = grid[0], grid[1]
+
+    lm = np.vstack((ll.flatten(), mm.flatten())).T
+    # Create input variables
+    coords = np.empty((nsrc, 4, 2)).astype(FT)
```

Comment:
Coords (lm coordinates) shouldn't need a correlation dimension.

```suggestion
    lm = np.empty((nsrc, 2)).astype(FT)
```